src/Components/Department.jsx

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+
+const Department = () => {
+  // eslint-disable-next-line no-unused-vars
+  const [department, setDepartment] = useState([]);
+
+  useEffect(() => {
+    fetch("http://localhost:3000/department/all")
+      .then((res) => res.json())
+      .then((data) => {
+        if (data.Status) {
+          setDepartment(data.data);
+        } else {
+          console.log(data.message);
+        }
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }, []);
+
+  function handleDelete(id) {
+    fetch(`http://localhost:3000/department/${id}/delete`, {
+      method: "DELETE",
+    })
+      .then((res) => res.json())
+      .then((data) => {
+        if (data.Status) {
+          window.location.reload();
```

Comment:
Okay Sir, but wouldn't it create an infinite rerendering due to state change?
is it okay to do so?

Reply:
You are correct about that. You should instead update the state to remove the deleted item instead of refreshing the page. Here is what it might look like: ```jsx 
        if (data.Status) {
          setDepartment(
            prevDepartments.filter((d) => d.id !== id)
          );
        } else {
          console.log(data.message);
        }
  }

```